WordPress/Classes/ViewRelated/Post/PublishSettingsViewController.swift

```diff
@@ -0,0 +1,179 @@
+    }
+
+    private var state: State
+    let timeZone: OffsetTimeZone?
+
+    let title: String?
+
+    private let post: AbstractPost
+
+    init(post: AbstractPost) {
+        if let dateCreated = post.dateCreated {
+            if post.hasFuturePublishDate() {
+                state = .scheduled(dateCreated)
+            } else {
+                state = .published(dateCreated)
+            }
+        } else {
+            state = .immediately
+        }
+
+        self.post = post
+
+        title = post.postTitle
+
+        if let gmtOffset = post.blog.settings?.gmtOffset {
+            timeZone = OffsetTimeZone(offset: gmtOffset.floatValue)
+        } else {
+            timeZone = nil
+        }
+    }
+
+    fileprivate var cells: [PublishSettingsCell] {
+        switch state {
+        case .published, .immediately:
+            return [PublishSettingsCell.dateTime]
+        case .scheduled:
+            return PublishSettingsCell.allCases
+        }
+    }
+
+    var date: Date? {
+        switch state {
+        case .scheduled(let date), .published(let date):
+            return date
+        case .immediately:
+            return nil
+        }
+    }
+
+    mutating func setDate(_ date: Date?) {
+        if let date = date {
+            state = .scheduled(date)
+        } else {
+            state = .immediately
+        }
+
+        post.dateCreated = date
+    }
+}
+
+@objc class PublishSettingsController: NSObject, SettingsController {
+
+    @objc class func viewController(post: AbstractPost) -> ImmuTableViewController {
+        let controller = PublishSettingsController(post: post)
+        let viewController = ImmuTableViewController(controller: controller)
+        return viewController
+    }
+
+    var noticeMessage: String?
+
+    let title = NSLocalizedString("Publish", comment: "Title for the publish settings view")
+
+    var immuTableRows: [ImmuTableRow.Type] {
+        return [
+            EditableTextRow.self
+        ]
+    }
+
+    static var dateFormatter: DateFormatter = {
+        let dateFormatter = DateFormatter()
+        dateFormatter.dateStyle = .long
+        dateFormatter.timeStyle = .short
+        return dateFormatter
+    }()
+
+    private var viewModel: PublishSettingsViewModel
+
+    init(post: AbstractPost) {
+        viewModel = PublishSettingsViewModel(post: post)
+
+        let siteTimezoneOffset = viewModel.timeZone?.gmtOffset ?? 0
+        let deviceTimezoneOffset = TimeZone.current.secondsFromGMT()/60/60
+        if siteTimezoneOffset != Float(deviceTimezoneOffset) {
+            noticeMessage = NSLocalizedString("Site time zone differs from device time zone", comment: "Notice that time zones are different when scheduling")
+        }
+    }
+
+    func tableViewModelWithPresenter(_ presenter: ImmuTablePresenter) -> ImmuTable {
+        return mapViewModel(viewModel, presenter: presenter)
+    }
+
+    func refreshModel() {
+        // Don't need to refresh the model
```

Comment:
Can we add more comments here explaining why this method can't be removed? Of course, the first thing I did was remove it, and _now_ I see why it has to be there.  😄

Reply:
Sure, I can flesh this comment out. It's a required method for `SettingsController` that isn't really used here.